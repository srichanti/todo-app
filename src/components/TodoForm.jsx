import React from 'react'
import styled from 'styled-components'

const FormInput = styled.input`
  width: 235px;
  outline: none;
  font-size: 13px;
  padding-top: 7px;
  padding-bottom: 7px;
  padding-left: 10px;
`

const TodoForm = () => (
  <form>
    <FormInput placeholder="Enter new todo" />
  </form>
)

export default TodoForm
