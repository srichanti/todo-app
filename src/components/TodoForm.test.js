import React from 'react'
import { shallow, mount } from 'enzyme'
import TodoForm from './TodoForm'

describe('TodoForm', () => {
  let shallowComponent
  let mountComponent
  let handleSubmit
  let handleOnchangeInput
  let currentTodo

  beforeEach(() => {
    handleSubmit = jest.fn()
    handleOnchangeInput = jest.fn()
    currentTodo = ''
    shallowComponent = shallow(<TodoForm />)
    mountComponent = mount(
      <TodoForm handleSubmit={handleSubmit} handleOnchangeInput={handleOnchangeInput} currentTodo={currentTodo} />
    )
  })
  it('renders correctly', () => {
    expect(shallowComponent).toMatchSnapshot()
  })

  it('TodoForm requires handleSubmit prop', () => {
    expect(mountComponent.props().handleSubmit).toBeDefined()
  })

  it('TodoForm requires currentTodo prop', () => {
    expect(mountComponent.props().currentTodo).toBeDefined()
  })
  it('TodoForm requires handleOnchange prop', () => {
    expect(mountComponent.props().handleOnchangeInput).toBeDefined()
  })

  it('Event handler should trigger', () => {
    const submitEle = mountComponent.find('form')
    submitEle.simulate('submit')
    expect(handleSubmit).toHaveBeenCalled()
  })
})
