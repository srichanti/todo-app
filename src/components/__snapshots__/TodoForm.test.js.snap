// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoForm Component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodoForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <styled.input
        placeholder="Enter new todo"
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "placeholder": "Enter new todo",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <styled.input
          placeholder="Enter new todo"
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "placeholder": "Enter new todo",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
